### PR TITLE
fix(node): Correct the signature of finished() in stream/promises

### PR DESCRIPTION
**Pull request by hkleungai**
The `cleanup` option is added to the promise version of `stream.finished(stream[, options])` since v19.1.0, v18.13.0. DT type should hence reflect this change.
- See [the documentation](https://nodejs.org/api/stream.html#streamfinishedstream-options).
- Resolve [DT discussion 70164](https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/70164) and [DT issue 70621](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/70621).

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>

### CODE REVIEW

**Comment by hkleungai** [comment]
Note: CI build fails due to downstream package. I am currently asking author of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70868 to see if he can help with this.

File: types/node/v20/stream.d.ts [hunk 1]
```diff
@@ -1392,14 +1392,12 @@ declare module "stream" {
...
-         * A readable and/or writable stream/webstream.
-         *
          * A function to get notified when a stream is no longer readable, writable
          * or has experienced an error or a premature close event.
          *
          * ```js
-         * import { finished } from 'node:stream';
-         * import fs from 'node:fs';
+         * const { finished } = require('node:stream');
+         * const fs = require('node:fs');
```
Comment (jakebailey) [inline types/node/v20/stream.d.ts:1400]: Same here.

File: types/node/stream.d.ts [hunk 2]
```diff
@@ -1411,14 +1411,12 @@ declare module "stream" {
...
-         * A readable and/or writable stream/webstream.
-         *
          * A function to get notified when a stream is no longer readable, writable
          * or has experienced an error or a premature close event.
          *
          * ```js
-         * import { finished } from 'node:stream';
-         * import fs from 'node:fs';
+         * const { finished } = require('node:stream');
+         * const fs = require('node:fs');
```
Comment (jakebailey) [inline types/node/stream.d.ts:1419]: Please do not make unrelated modifications.

**Review by jakebailey — CHANGES_REQUESTED** [review]

File: types/node/stream.d.ts (hunk 2 above)
Comment (hkleungai) [reply types/node/stream.d.ts]: Done.

File: types/node/v20/stream.d.ts (hunk 1 above)
Comment (hkleungai) [reply types/node/v20/stream.d.ts]: Done.

**Review by jakebailey — APPROVED** [review]

**Comment by hkleungai** [comment]
Ready to merge
